pkg/controllers/user/certsexpiration/certsexpiration.go

```diff
@@ -161,3 +169,32 @@ func getCertExpiration(c string) (v3.CertExpiration, error) {
 		ExpirationDate: date.Format(time.RFC3339),
 	}, nil
 }
+
+//deleteUnusedCerts removes unused certs and cleans up kubelet certs when GenerateServingCertificate is disabled
+func deleteUnusedCerts(certsExpInfo map[string]v3.CertExpiration, rancherKubernetesEngineConfig *v3.RancherKubernetesEngineConfig) {
+	logrus.Info("Checking and deleting unused certificates")
+	unusedCerts := make(map[string]bool)
+	for k := range certsExpInfo {
+		if strings.HasPrefix(k, pki.EtcdCertName) || strings.HasPrefix(k, pki.KubeletCertName) {
+			unusedCerts[k] = true
+		}
+	}
+	etcdHosts := hosts.NodesToHosts(rancherKubernetesEngineConfig.Nodes, services.ETCDRole)
+	allHosts := hosts.NodesToHosts(rancherKubernetesEngineConfig.Nodes, "")
+	for _, host := range etcdHosts {
+		etcdName := pki.GetCrtNameForHost(host, pki.EtcdCertName)
+		delete(unusedCerts, etcdName)
+	}
+	if pki.IsKubeletGenerateServingCertificateEnabledinConfig(rancherKubernetesEngineConfig) {
+		for _, host := range allHosts {
+			kubeletName := pki.GetCrtNameForHost(host, pki.KubeletCertName)
+			delete(unusedCerts, kubeletName)
+		}
+	}
+	if unusedCerts != nil {
```

Comment:
This check doesn't do anything as you made the map above. If there is nothing in the list the range is a no-op.